doozer/doozerlib/backend/rebaser.py

```diff
@@ -872,31 +872,57 @@ def _add_build_repos(self, dfp: DockerfileParser, metadata: ImageMetadata, dest_
 
             # The value we will set REMOTE_SOURCES_DIR to.
             remote_source_dir_env = '/tmp/cachito-emulation'
-
-            for npm_entry in metadata.config.cachito.packages.npm:
-                # cachito creates an .npmrc file in the "npm" package manager paths. This
-                # used to cause NPM to use an NPM server, supplied by cachito,
-                # which only supplied the component's NPM deps.
-                # See https://github.com/containerbuildsystem/cachito/blob/fe2ef761d7d8169597166e9f92ee8ef3d62a091c/README.md .
-                # Dockerfiles may reference this file. We create it in emulation mode, but do
-                # not populate it. cachi2 works fundamentally differently
-                # by re-writing package-lock.json to reference local files.
-                # https://github.com/hermetoproject/cachi2/blob/2235854bfc5cb76ea168cb04d50be16efce9fa0a/docs/npm.md
-                if 'path' in npm_entry:
+            pkg_managers = metadata.config.content.source.pkg_managers
+
+            if "npm" in pkg_managers:
+                flag = False
+                for npm_entry in metadata.config.cachito.packages.npm:
+                    # cachito creates an .npmrc file in the "npm" package manager paths. This
+                    # used to cause NPM to use an NPM server, supplied by cachito,
+                    # which only supplied the component's NPM deps.
+                    # See https://github.com/containerbuildsystem/cachito/blob/fe2ef761d7d8169597166e9f92ee8ef3d62a091c/README.md .
+                    # Dockerfiles may reference this file. We create it in emulation mode, but do
+                    # not populate it. cachi2 works fundamentally differently
+                    # by re-writing package-lock.json to reference local files.
+                    # https://github.com/hermetoproject/cachi2/blob/2235854bfc5cb76ea168cb04d50be16efce9fa0a/docs/npm.md
+                    if 'path' in npm_entry:
+                        app_path = gomod_deps_path.joinpath('app')
+                        app_path = app_path.joinpath(npm_entry['path'])
+                        app_path.mkdir(parents=True, exist_ok=True)
+                        app_path.joinpath('.npmrc').touch(exist_ok=True)
+                        flag = True
+
+                if not flag:
+                    # In some cases, in image config, metadata.config.content.source.pkg_managers will be populated
+                    # but metadata.config.cachito.packages.npm will be emtpy. In that case, we assume that the path
+                    # is the current directory
                     app_path = gomod_deps_path.joinpath('app')
-                    app_path = app_path.joinpath(npm_entry['path'])
+                    app_path = app_path.joinpath(".")
```

Comment:
joining `.` is actually a no-op.

Reply:
Thanks [updated](https://github.com/openshift-eng/art-tools/pull/1412/commits/6568247ab7d2c7df0632e7443dc7ab7d672d6924)